src/netbase.cpp

```diff
@@ -728,6 +724,13 @@ bool SetSocketNonBlocking(SOCKET& hSocket, bool fNonBlocking)
     return true;
 }
 
+bool SetSocketNoDelay(SOCKET& hSocket)
+{
+    int set = 1;
+    int rc = setsockopt(hSocket, IPPROTO_TCP, TCP_NODELAY, (const char*)&set, sizeof(int));
```

Comment:
Indeed, seems like the WIN32 check was unnecessary, every pointer can be passed to a `void*` parameter including `char*` so the redundant cast doesn't matter.